mitosheet/mitosheet/api/get_parameterizable_params.py

```diff
@@ -28,3 +28,27 @@ def get_parameterizable_params(params: Dict[str, Any], steps_manager: StepsManag
                 all_parameterizable_params.extend(parameterizable_params)
 
         return all_parameterizable_params
+
+# This function is used to get the parameterizable params for the UI in streamlit. 
+# It returns a list of ParamMetadata objects, which are defined in mitosheet/types.py
+# The ParamMetadata object includes various information that a streamlit developer might
+# need to display the parameterizable params in the UI.
+def get_parameterizable_params_metadata(steps_manager: StepsManagerType) -> List[ParamMetadata]:
+        # First, get all the parameterizable params and the function params
+        all_parameterizable_params = get_parameterizable_params({}, steps_manager)
+        function_params: Dict[ParamName, ParamValue] = get_final_function_params_with_subtypes_turned_to_parameters(steps_manager, 'all')
+        param_names = list(function_params.keys())
```

Comment:
Ah -- one thing here is that these may not be ordered (dicts, by default, are not). I think you can fix this just by changing the `get_final_function_params_with_subtypes_turned_to_parameters` function to return an ordered dict!